runtime/lua/vim/lsp.lua

```diff
@@ -1110,6 +1114,26 @@ function lsp.buf_notify(bufnr, method, params)
   return resp
 end
 
+
+--@private
+--- Get the details of the completion item
+function lsp._resolve_completion_detail()
+  local bufnr = resolve_bufnr()
+  local completed_item_var = vim.api.nvim_get_vvar('completed_item')
+  local item = completed_item_var.user_data.lsp.completion_item
+  local lnum = item.data.line
+  lsp.buf_request(bufnr, 'completionItem/resolve', item, function(err, _, result)
+    if err or not result then return end
+    if result.additionalTextEdits then
+      local edits = vim.tbl_filter(
+        function(x) return x.range.start.line ~= (lnum - 1) end,
+        result.additionalTextEdits
+      )
```

Comment:
why do we have to do the filter?

Reply:
I think this logic came originally from me and somehow made it into nvim-completion. The reason for it was that in my version I did a snippet expansion right after using `item.textEdit.newText`, and if the `additionalTextEdits` affect the current line it would affect the cursor position and mess up the snippet expansion positioning. 

The spec also says this:

```
Additional text edits should be used to change text unrelated to the
	 * current cursor position (for example adding an import statement at the
	 * top of the file if the completion item will insert an unqualified type).
```